DSharpPlus/Entities/Interaction/DiscordHttpInteraction.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using DSharpPlus.Net.Abstractions;
+using DSharpPlus.Net.Serialization;
+using Newtonsoft.Json;
+
+namespace DSharpPlus.Entities;
+
+public class DiscordHttpInteraction : DiscordInteraction
+{
+    [JsonIgnore]
+    internal readonly TaskCompletionSource taskCompletionSource = new();
+    
+    [JsonIgnore]
+    internal byte[] response;
+
+    internal bool Cancel() => this.taskCompletionSource.TrySetCanceled();
+
+    internal async Task<byte[]> GetResponseAsync()
+    {
+        await this.taskCompletionSource.Task;
+
+        return this.response;
+    }
+    
+    public override Task CreateResponseAsync(DiscordInteractionResponseType type, DiscordInteractionResponseBuilder? builder = null)
+    {
+        if (this.ResponseState is not DiscordInteractionResponseState.Unacknowledged)
```

Comment:
There should also be a check for if the TCS has been canceled, since as it stands right now, this will throw an exception if for any reason the CT is cancelled, though that exception probably won't contain very useful information (Invalid op with "TCS has already been set" - which doesn't mean much to users nor developers here imo)